### PR TITLE
feat: use and customise Carbon light theme

### LINKED ISSUE #965: ❤️👍👍👍   Use Carbon themes and customize it to match visual identity
Since all of the pages have moved to the light theme, we need to modify the `community-theme.scss` file to match our visual branding.

This work is the follow-up for the suggestion mentioned [here](https://github.com/Qiskit/qiskit.org/pull/951#discussion_r495666584).

### PR DESCRIPTION
**Pull request by eddybrando**
This PR refactors the current Carbon theme in order to extend the **white** and not the **gray 100** version.

Some other improvements were made to the way we import CSS styles and variables to avoid, for instance, repeating the import of the _copy_ styles files in various components.

The special styling for the "metal" page has algo been simplified.

As part of the refactory, we're also now using the appropriate Carbon color tokens when possible.

---

Closes #965 

Addresses the FOUT part of #1569

### CODE REVIEW

**Comment by y4izus** [comment]
@eddybrando do you think you could resolve this issue #1366 while doing this refactor?

File: nuxt.config.ts [hunk 1]
```diff
@@ -131,7 +124,8 @@ const config: NuxtConfig = {
     ['@nuxt/typescript-build', {
       typeCheck: true,
       ignoreNotFoundWarnings: true
-    }]
+    }],
+    '@nuxtjs/style-resources',
```
Comment (y4izus) [inline nuxt.config.ts]: Per offline conversations, we move the `@nuxtjs/stype-resources` into `buildModules` to follow the [instructions on the setup](https://github.com/nuxt-community/style-resources-module/#setup)

**Review by y4izus — APPROVED** [review]
Great refactor!! this is very useful!! thank you very much Eddy!

**Comment by y4izus** [comment]
I noticed that some styles are broken, could you take a look at them, please?
- The hover effect on the "copy" button doesn't work
Captura de pantalla 2021-04-20 a las 9 42 59

- When selecting an option on the quick start, it turns blue
Captura de pantalla 2021-04-20 a las 9 44 38

- Also on the quick start, when clicking on the accordion, the blue appears:
Captura de pantalla 2021-04-20 a las 9 45 33

- The menu link is not purple when it's active because we are in that page. On mobile menu it works
Captura de pantalla 2021-04-20 a las 9 46 46

- On the overview page, when clicking on the accordion, again the blue:
Captura de pantalla 2021-04-20 a las 9 48 19

- Checkboxes has not the proper styles (refactor vs production)(learn page):
Captura de pantalla 2021-04-20 a las 9 51 33
Captura de pantalla 2021-04-20 a las 9 52 41

- Blue instead of purple (learn page):
Captura de pantalla 2021-04-20 a las 9 53 09

- Blue instead of purple (learn page):
Captura de pantalla 2021-04-20 a las 9 53 50

- Blue instead of purple (events page) (it's the same for all the filters):
Captura de pantalla 2021-04-20 a las 9 55 11

- Blue instead of purple (events page) (refactor vs production)
Captura de pantalla 2021-04-20 a las 9 57 19
Captura de pantalla 2021-04-20 a las 9 57 48

- Blue instead of purple  (events page) (refactor vs production)
Captura de pantalla 2021-04-20 a las 9 58 23
Captura de pantalla 2021-04-20 a las 9 58 40

- Same on the advocate's page for the filters

File: nuxt.config.ts (hunk 1 above)
Comment (eddybrando) [reply nuxt.config.ts]: @delapuente FYI

**Comment by eddybrando** [comment]
@y4izus great catches! Thanks a lot for the detailed review. I just checked and fixed all of the mentioned bugs and fixed #1366 as well.

As we discussed offline and you pointed out, this is a prime example of our need for an automated visual regression testing process, that could help us spare a lot of time in this kind of reviews and avoid regressions.

**Review by y4izus — APPROVED** [review]
Now it's perfect Eddy!! thank you very much for the efforts!

**Pull request merged**